src/Microsoft.AspNetCore.SystemWebAdapters/Caching/Cache.cs

```diff
@@ -1,24 +1,152 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-namespace System.Web.Caching
+using System.Collections;
+using System.Runtime.Caching;
+
+namespace System.Web.Caching;
+
+public delegate void CacheItemRemovedCallback(string key, object value, CacheItemRemovedReason reason);
+public delegate void CacheItemUpdateCallback(string key, CacheItemUpdateReason reason, out object? expensiveObject, out CacheDependency? dependency, out DateTime absoluteExpiration, out TimeSpan slidingExpiration);
+
+public sealed class Cache : IEnumerable
 {
-    public sealed class Cache
+    private readonly ObjectCache _cache;
```

Comment:
Yes, the main point is more a bridge rather than necessarily exposing the same extensibility. That said, if it's a used pattern, it's something we could add.

@Tratcher what are your thoughts here? I can move things to go through a CacheStoreProvider to enable similar extension aspect if we want. This is something we could add later without breaking changes, so not something we probably need to do initially, but this is good to know.

Reply:
I don't think `CacheStoreProvider` is used by anyone since we never really announced it. It's there. Somebody could use it. But we never blogged about it or posted any guidance or examples, so probably nobody is using it. I guess my question was less about `CacheStoreProvider` specifically - which isn't really special or well thought out... its basically just a simple exposure of the API needed for System.Web.Caching... which is kind of an odd cache. I was more curious about whether the choice to use `ObjectCache` was intentional or just a convenience.

Reply:
Yes, the use of `ObjectCache` was intentional. By default we'll just use MemoryCache, but it allows someone to swap it out for their own implementation on ObjectCache if they want to control the behavior at all.